### ISSUE TITLE
Traefik unable to obtain ACME certificate for domains "traefik.baikal."

### ISSUE THREAD

**Issue by SamuelBlickle** [ckulka/baikal-docker#28]
Hi,

when I try to start up the containers for baikal and traefik, docker-compose throws the following error message:
`traefik_1  | time="2020-11-05T14:50:06Z" level=error msg="Unable to obtain ACME certificate for domains \"traefik.baikal.\" detected thanks to rule \"Host:traefik.baikal.\" : cannot obtain certificates: acme: Error 400 - urn:ietf:params:acme:error:rejectedIdentifier - Error creating new order :: Cannot issue for \"traefik.baikal.\": Domain name ends in a dot"`

My compose file looks exactly like the `docker-compose.ssl.yaml` example with the `traefik.frontend.rule` adjusted to my host and the `acme.json` created in a folder.

Any ideas what's wrong here?
Thank you very much!

BR
Samuel Blickle

**Comment by ckulka** [comment]
Hi @SamuelBlickle ,

this fails because the domain name is not valid for ACME, i.e. the automated certificate generation, because it ends with a dot (you can also see this at the end of the error message).

To fix this, just remove the `.` from the `traefik.frontend.rule` and you will be fine.

But wait, there is more: **after removing the dor the error will be gone, but it will continue to fail.**

Why? Let's Encrypt (which we use here to generate the certificate) must be able to reach your Baikal instance under the `traefik.frontend.rule`, i.e. under <https://traefik.baikal>. Since that URL is not accessible from the internet, Let's Encrypt cannot verify your domain and errors out.

How to fix this? You'll need to use a domain under which Let's Encrypt can reach your Baikal instance from the internet, e.g. `baikal.samuelblickle.de`.

Let me know if you don't want to put your Baikal instance on the internet, [there's also a way to get this to work](https://doc.traefik.io/traefik/v1.6/configuration/acme/#dnschallenge), but it depends on your DNS registrar. Here's a list of compatible ones: <https://doc.traefik.io/traefik/v1.6/configuration/acme/#provider>.

**Comment by SamuelBlickle** [comment]
Hi @ckulka ,

thank you very much for your reply.
However this unfortunately doesn't answer my question, but I think that's my fault. I should have stated what's the weird point of my problem:

I have of course set up the `traefik.frontend.rule` to something like `baikal.mydomain.tld` (and configured the DNS settings accordingly), but docker-compose throws exactly the error message shown in my initial question (with `\"Host:traefik.baikal.\"`).
Therefore I am not quite sure what causes this behavior or what to change. Also, my domain obviously does not end with a `.` so the problem must be caused by something that is not related to the docker-compose file I'd say.

However, I also noticed that the containers do indeed start up perfectly fine. They even set up the SSL/TLS certificate for my specified domain. I did not notice this at first, because I interpreted the output of docker-compose as "something's wrong, setup failed at this step" and somehow never got the idea of actually testing it. 

So the error shown above does get thrown but everything works as it should, as far as I can see. That's a bit strange.
